### ISSUE TITLE
Eager loaded relationships with CamelCase method naming

### ISSUE THREAD

**Issue by AadilC** [yajra/laravel-datatables#105]
Hi,

Eager loading __CamelCased__ relationships on eloquent models returns no results in the result set in addition to changing the variable name to __snake_case__.

This basic query
```php
    $products = Product::withTrashed()
            ->with('variantsCount')
            ->select('*');
    
    return  Datatables::of($products)->make(true);
```
produces this output in the result set
```php
... "variants_count":null, ...
```

as such no variation in the __name__ field produces any output to the datatable
```php
->addColumn(['data' => 'variantsCount', 'name' => 'variantst', 'title' => 'Total Variants'])
```

The query is valid and returns results when used elsewhere.

Kind Regards.

**Comment by yajra** [comment]
@AadilC, i just tested this issue and it seems that the problem is on Laravel Framework and not in the package. Try returning the query directly like below. Maybe you should report this to the framework instead. Good catch! (Tested on Laravel 5)

This works!
```php
        $posts = Post::with('user')->select('*');
        return $posts->get();
```

This doesn't work!
```php
        $posts = Post::with('userData')->select('*');
        return $posts->get();
```

**Comment by AadilC** [comment]
@yajra thank you for your response.

Just to note, accessing the same relationship attribute worked on a previous version of __yajra/laravel-datatables__ if it was accessed as a snake_cased identifier __variants_count__
```php
->addColumn(['data' => 'variants_count', 'name' => 'variants', 'title' => 'Total Variants'])
```


I'm am running
```console
Laravel Framework version 5.1.6 (LTS)
```

when executing the query

```php
$products = Product::withTrashed()
            ->with('variantsCount')
            ->select('*')
            ->first();

dd($products);
```
I get the following output
```php
Collection {#407 ▼
  #items: array:1 [▼
    0 => Product {#395 ▼
      #table: "products"
     ...
      #relations: array:1 [▼
       "variantsCount" => Variant {#408 ▼
          #table: "variants"
          ...
          #attributes: array:2 [▼
            "product_id" => 1
            "value" => 2
          ]
          ...
          +exists: true
          #forceDeleting: false
        }
      ]
     ...
      +exists: true
      #forceDeleting: false
    }
  ]
}
```

the __variantsCount__  eager-loaded relationship is present is included in the __relations__ array of the output.

The output does produce results but it seems must be accessed as the original __camelCase__ value as per https://github.com/laravel/framework/issues/3419.

Also changing the method signature from
```php
public function variantsCount()
    {
        return $this->hasOne(Variant::class)
                    ->selectRaw('product_id, count(*) as value')
                    ->groupBy('product_id');
    }
```
to 
```php
public function variants_count()
    {
        return $this->hasOne(Variant::class)
                    ->selectRaw('product_id, count(*) as value')
                    ->groupBy('product_id');
    }
```

works but is not an ideal solution. should it not access the relationship as it is returned by Eloquent?

**Comment by yajra** [comment]
Thanks for noting that it works on previous version. Will try this issue on my L5.1 app and let you know what I may find.

**Comment by yajra** [comment]
@AadilC, I think I found the problem. Eager loading for `hasOne` and `hasMany` seems to be buggy. In this regard, can you please test this solution. On `Datatables\Helper.php`, please update `convertToArray` function to:

```php
public static function convertToArray($row)
    {
        $data = $row instanceof Arrayable ? $row->toArray() : (array) $row;
        foreach (array_keys($data) as $key) {
            if (is_object($data[$key])) {
                $data[$key] = $row->$key;
            } else if (is_array($data[$key])) {
                $data[$key] = static::convertToArray($data[$key]);
            } else {
                $data[$key] = $row[$key];
            }
        }

        return $data;
    }
```

Have tested it on my local machine but it would be great if you test this patch before I post and released it. Thanks a lot!

**Comment by yajra** [comment]
Have posted the patch on `dev-master`. Will tag it for release once you confirmed the issue was fixed. Thanks and sorry for the inconvenience.

**Comment by AadilC** [comment]
@yajra Kudos the patch works. I really appreciate the help and your package has been an invaluable time saver :smiley:

**Comment by yajra** [comment]
great.. i will tag for a release. thanks for confirming!

**Comment by yajra** [comment]
should be fixed on v5.5.6. thanks a lot! 

FYI: I made some changes in the function and removed the unnecessary else condition.

```php
public static function convertToArray($row)
    {
        $data = $row instanceof Arrayable ? $row->toArray() : (array) $row;
        foreach (array_keys($data) as $key) {
            if (is_object($data[$key])) {
                $data[$key] = $row->$key;
            } else if (is_array($data[$key])) {
                $data[$key] = static::convertToArray($data[$key]);
            }
        }

        return $data;
    }
```
